### PR TITLE
[Spring Cleanup] Remove Spring dependencies in the IDVP management API

### PR DESCRIPTION
**Pull request by NipuniBhagya**
## Related Issue(s)

- https://github.com/wso2/product-is/issues/21159

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.idv.provider/org.wso2.carbon.identity.api.server.idv.provider.common/src/main/java/org/wso2/carbon/identity/api/server/idv/provider/common/IdentityVerificationServiceHolder.java [hunk 1]
```diff
@@ -18,14 +18,20 @@
...
 /**
  * Service holder class for identity verification Rest API.
  */
 public class IdentityVerificationServiceHolder {
 
-    private static IdVProviderManager idVProviderManager;
+    private IdentityVerificationServiceHolder() {};
+
+    private static class IdVProviderManagerHolder {
+        static final IdVProviderManager SERVICE = (IdVProviderManager) PrivilegedCarbonContext
```
Comment (lashinijay) [inline components/org.wso2.carbon.identity.api.server.idv.provider/org.wso2.carbon.identity.api.server.idv.provider.common/src/main/java/org/wso2/carbon/identity/api/server/idv/provider/common/IdentityVerificationServiceHolder.java:32]:
```suggestion

        static final IdVProviderManager SERVICE = (IdVProviderManager) PrivilegedCarbonContext
```

File: components/org.wso2.carbon.identity.api.server.idv.provider/org.wso2.carbon.identity.api.server.idv.provider.v1/src/main/java/org/wso2/carbon/identity/api/server/idv/provider/v1/impl/IdvProvidersApiServiceImpl.java [hunk 2]
```diff
@@ -37,8 +37,15 @@
  */
 public class IdvProvidersApiServiceImpl implements IdvProvidersApiService {
 
-    @Autowired
-    IdVProviderService idVProviderService;
+    private final IdVProviderService idVProviderService;
+
+    public IdvProvidersApiServiceImpl() {
+        try {
```
Comment (lashinijay) [inline components/org.wso2.carbon.identity.api.server.idv.provider/org.wso2.carbon.identity.api.server.idv.provider.v1/src/main/java/org/wso2/carbon/identity/api/server/idv/provider/v1/impl/IdvProvidersApiServiceImpl.java:43]:
```suggestion

        try {
```

File: components/org.wso2.carbon.identity.api.server.idv.provider/org.wso2.carbon.identity.api.server.idv.provider.common/src/main/java/org/wso2/carbon/identity/api/server/idv/provider/common/IdentityVerificationServiceHolder.java (hunk 1 above)
Comment (lashinijay) [reply components/org.wso2.carbon.identity.api.server.idv.provider/org.wso2.carbon.identity.api.server.idv.provider.common/src/main/java/org/wso2/carbon/identity/api/server/idv/provider/common/IdentityVerificationServiceHolder.java:32]: will address this in a separate PR, as this needs to be fixed in all the components

File: components/org.wso2.carbon.identity.api.server.idv.provider/org.wso2.carbon.identity.api.server.idv.provider.v1/src/main/java/org/wso2/carbon/identity/api/server/idv/provider/v1/impl/IdvProvidersApiServiceImpl.java (hunk 2 above)
Comment (lashinijay) [reply components/org.wso2.carbon.identity.api.server.idv.provider/org.wso2.carbon.identity.api.server.idv.provider.v1/src/main/java/org/wso2/carbon/identity/api/server/idv/provider/v1/impl/IdvProvidersApiServiceImpl.java:43]: will address this in a separate PR, as this needs to be fixed in all the components

**Comment by lashinijay** [comment]
Comments will be addressed with https://github.com/wso2/identity-api-server/pull/756

**Review by nilasini — APPROVED** [review]

**Pull request merged**